Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -2898,6 +2902,107 @@ bool Element::has_paint_containment() const
     return false;
 }
 
+int Element::number_of_owned_list_items()
+{
+    auto number_of_owned_li_elements = 0;
+    for_each_child_of_type<DOM::Element>([&](auto& child) {
+        if (child.list_owner() == this) {
+            number_of_owned_li_elements++;
+        }
+        return IterationDecision::Continue;
+    });
+    return number_of_owned_li_elements;
+}
+
+// https://html.spec.whatwg.org/#list-owner
+Element* Element::list_owner()
+{
+    // Any element whose computed value of 'display' is 'list-item' has a list owner, which is determined as follows:
+    if (!computed_properties()->display().is_list_item())
+        return nullptr;
+
+    // FIXME: 1. If the element is not being rendered, return null; the element has no list owner.
+
+    // 2. Let ancestor be the element's parent.
+    auto* ancestor = parent();
+
+    // 3. If the element has an ol, ul, or menu ancestor, set ancestor to the closest such ancestor element.
+    for (auto* node = ancestor; node; node = node->parent()) {
+        if (is<HTML::HTMLOListElement>(node) || is<HTML::HTMLUListElement>(node) || is<HTML::HTMLMenuElement>(node)) {
+            ancestor = node;
+            break;
+        }
+    }
+
+    // 4. Return the closest inclusive ancestor of ancestor that produces a CSS box.
+    for (; ancestor; ancestor = ancestor->parent()) {
+        if (is<Element>(ancestor)) {
+            return static_cast<Element*>(ancestor);
+        }
+    }
+
+    // Spec-Note: Such an element will always exist, as at the very least the document element will always produce a CSS box.
+    VERIFY_NOT_REACHED();
+}
+
+// https://html.spec.whatwg.org/#ordinal-value
+int Element::ordinal_value()
+{
+    // NOTE: The spec provides an algorithm to determine the ordinal value of each element owned by a given list owner.
+    //       However, we are only interested in the ordinal value of this element.
+
+    // FIXME: 1. Let i be 1.
+
+    // 2. If owner is an ol element, let numbering be owner's starting value. Otherwise, let numbering be 1.
+    auto* owner = list_owner();
+    if (!owner) {
+        return 1;
+    }
+
+    auto numbering = 1;
+    auto reversed = false;
+    if (is<HTML::HTMLOListElement>(owner)) {
+        auto* ol_element = static_cast<HTML::HTMLOListElement*>(owner);
+        numbering = ol_element->starting_value();
+        reversed = ol_element->has_attribute(HTML::AttributeNames::reversed);
+    }
+
+    // FIXME: 3. Loop : If i is greater than the number of list items that owner owns, then return; all of owner's owned list items have been assigned ordinal values.
+    // FIXME: 4. Let item be the ith of owner's owned list items, in tree order.
+
+    owner->for_each_child_of_type<DOM::Element>([&](auto& item) {
+        if (item.list_owner() == owner) {
```

Comment:
Do you expect this condition to be true more than once? If not, you should `return IterationDecision::Break` at the end of this block.

Reply:
Yes, it is true for every item in that list. The algorithm iterates over all list items, updates the counter and returns the value if it reaches the list item whose ordinal value is being calculated.